src/main/kotlin/app/ui/userCreation/View/UserCreation.kt

```diff
@@ -0,0 +1,161 @@
+*  gives the user visual feedback through the recording process
+*
+* */
+class UserCreation : View("Creating User") {
+
+
+    private val mIcon = MaterialIconView(MaterialIcon.CLOSE, "25px")
+    private val viewModel : UserCreationViewModel  by inject ()
+    lateinit var recordButton: VBox
+    private val progressBar = ProgressBar()
+    val isRecording = viewModel.isRecording
+    val doneRecording = viewModel.doneRecording
+    var timer = Timer()
+
+    //initialize close button to be used in top of borderpane
+    val closeButton = button("CLOSE", mIcon) {
+        importStylesheet(ButtonStyles::class)
+        addClass(ButtonStyles.rectangleButtonDefault)
+
+        style {
+            alignment = Pos.CENTER
+            mIcon.fill = c("#CC4141")
+            effect = DropShadow(10.0, Color.GRAY)
+
+        }
+        action {
+            navHome()
+        }
+    }
+
+    override val root = borderpane {
+
+        style{
+            backgroundColor+= Color.WHITE
+        }
+
+        top {
+
+            //close button
+            hbox {
+                alignment = Pos.BOTTOM_RIGHT
+                add(closeButton)
+                style {
+                    alignment = Pos.BOTTOM_RIGHT
+                    paddingRight= 40.0
+                    paddingTop = 40.0
+                }
+            }
+
+        }
+        center{
+
+                isRecording.subscribeBy(
+                        onNext = {
+                            if (it == false) { // if isRecording set to false create new RecordButton()
+                                val newRecordButton = RecordButton(::onClickCallback, ::animationCompleted, ::stopClicked)
+                                newRecordButton.alignment = Pos.CENTER
+                                recordButton.replaceWith(newRecordButton)
+                                recordButton = newRecordButton
+                            }
+                        } ,
+
+                        onError = {
+                            println(it)
+                        }
+                )
+
+                doneRecording.subscribeBy(
+                        onNext = {
+                                if(it == true) { //done recording = true? then navigate to profile preview
+                                timer.schedule(timerTask {
+                                    Platform.runLater {
+                                        find(UserCreation::class).replaceWith(ProfilePreview::class,
+                                                transition = ViewTransition.Fade(0.3.seconds))
+                                    }
+                                }, 500)
+
+                            }
+                        },
+
+                onError = {
+                            println(it)
+                        }
+                )
+
+            // record button needs to be added here below bc the subjects haven't emitted anything yet
+
```

Comment:
...what?

Reply:
OH that's because they should be PublishSubjects, not BehaviorSubjects. 

http://reactivex.io/RxJava/javadoc/rx/subjects/BehaviorSubject.html
http://reactivex.io/RxJava/javadoc/rx/subjects/PublishSubject.html
http://reactivex.io/RxJava/javadoc/rx/subjects/Subject.html

Reply:
Ah ok. I thought that PublishSubjects behaved similarly because I had to do something similar in the rxdemo I was working on a couple weeks back. will go ahead and change it